test: fix tcp_close_while_connecting CI failures

bnoordhuis opened the pull request:
The expected error is UV_ECANCELED but the test tries to connect to what
is basically an arbitrary address in the expectation that no network
path exists, so UV_ENETUNREACH is an equally plausible outcome.

This commit undoes the change from commit e994000 ("test: make
tcp_close_while_connecting more resilient") because I don't think
the connection ever actually succeeds.

Refs: https://github.com/libuv/libuv/pull/1005
CI: https://ci.nodejs.org/job/libuv-test-commit/131/

iWuzHere commented:
LGTM

saghul commented:
LGTM.  I don't know what I was thinking when I "fixed" that test before :cry:  I blame the reviewer :stuck_out_tongue_closed_eyes:

santigimeno commented:
Should I close https://github.com/libuv/libuv/pull/1047 then?

cjihrig commented:
LGTM. This seems like a simpler fix than #1047 to me.

bnoordhuis commented:
I incorporated some of the changes from #1047 but I moved the RETURN_SKIP to after MAKE_VALGRIND_HAPPY.  CI: https://ci.nodejs.org/job/libuv-test-commit/132/

Looking at the test again, I noticed that uv_tcp_connect() reports UV_ENETUNREACH either synchronously or asynchronously.  That seems wrong, it should always be asynchronous.

santigimeno commented:
LGTM, I guess. I'll close https://github.com/libuv/libuv/pull/1047
